src/main/scala/loamstream/loam/intake/IntakeSyntax.scala

```diff
@@ -77,11 +82,11 @@ trait IntakeSyntax extends Interpolators {
       val tool = NativeTool {
         Files.writeLinesTo(dataListFile.path)(Iterator(dataFile.path.toString))
         Files.writeLinesTo(schemaListFile.path)(Iterator(schemaFile.path.toString))
-      }.out(dataListFile, schemaListFile)
+      }
       
       addToGraph(tool)
       
-      tool
+      tool.out(dataListFile, schemaListFile)
```

Comment:
Just trying to understand the change. The code is simple enough, but you went from adding the tool to the graph _with_ the transform action to adding before and then doing something?

Sorry, there's just so little context in the diff to properly "review" this.  I don't know what this accomplishes or what bug was fixed doing this?

Reply:
I should write a comment.  The tool needs to be added to the graph before calls to `.out()` have any effect.

Reply:
Does the tool assert if that happens in the tool code? For example:
```scala
def out(...) {
  assert(this.parentGraph.isDefined)
  // ...
}
```

Reply:
It doesn't; that's a good idea.